src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyGenerator.java

```diff
@@ -335,8 +335,7 @@ protected void engineInit(int keySize, SecureRandom random) {
         try {
             newSignificantKeySize = checkKeySize(mechanism, keySize, range);
         } catch (InvalidAlgorithmParameterException iape) {
-            throw (InvalidParameterException)
-                    (new InvalidParameterException().initCause(iape));
+            throw new InvalidParameterException("Invalid algorithm parameter", iape);
```

Comment:
I would change this to `throw new InvalidParameterException(iape)`. The detail message of the cause is more useful in this case.